src/hmm/hmm-topology.h

```diff
@@ -99,19 +99,28 @@ class HmmTopology {
     /// but may be different to enable us to hardwire sharing of state, and may be
     /// equal to \ref kNoPdf == -1 in order to specify nonemitting states (unusual).
     int32 pdf_class;
+    int32 self_loop_pdf_class;
 
     /// A list of transitions, indexed by what we call a 'transition-index'.
     /// The first member of each pair is the index of the next HmmState, and the
     /// second is the default transition probability (before training).
     std::vector<std::pair<int32, BaseFloat> > transitions;
 
-    explicit HmmState(int32 p): pdf_class(p) { }
+    explicit HmmState(int32 p): pdf_class(p),self_loop_pdf_class(p) { }
+    explicit HmmState(int32 p, int32 self_p) {
```

Comment:
better to rename these to pdf_class and self_loop_pdf_class [self-constructor still works even though it's the same as the variable names.]